### PR TITLE
feat: Add registration functionality and templates for login and register

### PR DESCRIPTION
**Pull request by sechaba-mache**

### CODE REVIEW

File: angular-holiday-planner/src/app/components/login/login.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<div class="login-form w-screen h-screen flex justify-center items-center">
+  <form [formGroup]="loginForm" class="w-[20rem] pb-6 bg-slate-100 rounded-xl text-center">
+    <h1 class="text-center mt-5 font-extrabold text-2xl">Login</h1>
+
+    <div class="form-content flex flex-wrap mt-5">
+      <label class="w-full text-xl font-bold m-3 ">Email:</label>
+      <input formControlName="email" type="email" placeholder="Email" class="ml-12 w-56 h-7 pl-3">
+
+      <label class="w-full text-xl font-bold m-3">Password:</label>
+      <input formControlName="password" type="password" placeholder="Password" class="ml-12 w-56 h-7 pl-3">
```
Comment (MihlaliMgwebi) [inline angular-holiday-planner/src/app/components/login/login.component.html:10]: a11y issues my arise from not adding a name attribute to your inputs. Please test using the WAVE extenstion

File: angular-holiday-planner/src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return this.loginForm.get("email")
+  }
+
+  get password() {
+    return this.loginForm.get("password")
+  }
+
+  submitLoginForm(){
+    const user: IUser = {
+      email: this.email?.value!,
```
Comment (MihlaliMgwebi) [inline angular-holiday-planner/src/app/components/login/login.component.ts:34]: (Gasp) is this a bang operator I see?

File: angular-holiday-planner/src/app/components/register/register.component.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+export class RegisterComponent {
+
+  constructor(private auth: AuthService, private router: Router) { }
+
+  register$: Observable<UserCredential> | null = null;
+
+  confirmPassword: string = '';
+
+  registerForm = new FormGroup({
+    // username: new FormControl('', [Validators.required]),
```
Comment (MihlaliMgwebi) [inline angular-holiday-planner/src/app/components/register/register.component.ts:24]: remove unused code

**Review by MihlaliMgwebi — CHANGES_REQUESTED** [review]
Saw your login page in person and it's cute. 

Please check my comments.

File: angular-holiday-planner/src/app/components/login/login.component.ts (hunk 2 above)
Comment (sechaba-mache) [reply angular-holiday-planner/src/app/components/login/login.component.ts:34]: Yes, that is a bang operator, reason being I know for a fact that these values are definitely going to be instantiated when the function is called. It is linked to a button click which is disabled unless all input fields have been entered and are valid so by the time you click on the button the values will exist

File: angular-holiday-planner/src/app/components/register/register.component.ts (hunk 3 above)
Comment (sechaba-mache) [reply angular-holiday-planner/src/app/components/register/register.component.ts:24]: Will do

File: angular-holiday-planner/src/app/components/login/login.component.html (hunk 1 above)
Comment (sechaba-mache) [reply angular-holiday-planner/src/app/components/login/login.component.html:10]: A11y will be accomodated for in a separate branch

File: angular-holiday-planner/src/app/components/login/login.component.ts (hunk 2 above)
Comment (MihlaliMgwebi) [reply angular-holiday-planner/src/app/components/login/login.component.ts:34]: Yes, I do love the fact that you disable your button if your form is invalid. Great use of directives!

However, never trust your DOM. Never trust your API. If they could definitely be instantiated they could never possibly be null and you wouldn't need a bang operator.

Add a guard clause to be safe.

**Review by MihlaliMgwebi — APPROVED** [review]
Love the conversation. I still don't approve of the bang operator. Other than that, looks really good. Request a re-review next time (as attached).

PS: I really love your use of structural directives for disabling buttons.

Screenshot 2023-03-31 at 11 22 44

File: angular-holiday-planner/src/app/components/login/login.component.ts (hunk 2 above)
Comment (sechaba-mache) [reply angular-holiday-planner/src/app/components/login/login.component.ts:34]: As per my reasoning above the values are definitely going to be instantiated before any operation is done with them so I don't think changes are required

**Pull request merged**